TestSystemCommands: Improve application startup detection to prevent sending a SIGTERM before signal handler is setup

enkiusz opened the pull request:
#### Summary

When running the YAML tests on my machine the `TestSystemCommands` testcase fails in 1/3 of the runs. I have tracked it down to steps 22 and 23 of the test plan where the app is factory reset twice in short order. It looks like the app does not have time to properly initialize signal handlers between first and second attempts as sending the second SIGTERM to stop it results in an error code -15 being returned. An example log tail from where this happens is provided below:

```
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [DMG] Endpoint 1, Cluster 0x0000_005D update version to 81d5316c
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [DMG] Endpoint 1, Cluster 0x0000_005D update version to 81d5316d
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [DMG] Endpoint 1, Cluster 0x0000_005D update version to 81d5316e
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [DMG] Endpoint 1, Cluster 0x0000_005D update version to 81d5316f
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [EVL] LogEvent event number: 0x0000000000000003 priority: 1, endpoint id:  0x1 cluster id: 0x0000_005D event id: 0x0 Epoch timestamp: 0x0000019A30D300D3
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [ZCL] Set number of supported temperature units to: 2
2025-10-29 16:35:26.258 ERROR   16:35:26.035 - APP  OUT  : [1761755726.035] [250644:250644] [DMG] Endpoint 0, Cluster 0x0000_002D update version to 49bd8791
2025-10-29 16:35:26.258 ERROR   16:35:26.065 - TEST OUT  : 		22. Running Factory Reset the default accessory
2025-10-29 16:35:26.258 ERROR   16:35:26.065 - TEST OUT  : ✓ 533.51ms
2025-10-29 16:35:26.258 ERROR   16:35:26.065 - TEST OUT  : 		    [PostProcessCheckType.IM_STATUS check] The test expects no error and no error occurred.
2025-10-29 16:35:26.258 ERROR   16:35:26.065 - TEST OUT  : 		***** Test Step 23 : Factory Reset the default accessory by key
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  : Traceback (most recent call last):
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/tests/chipyaml/chiptool.py", line 118, in <module>
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :     chiptool_py()
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 1157, in __call__
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :     return self.main(*args, **kwargs)
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 1078, in main
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :     rv = self.invoke(ctx)
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :          ^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.073 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 1434, in invoke
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :     return ctx.invoke(self.callback, **ctx.params)
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :     return __callback(*args, **kwargs)
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :     return f(get_current_context(), *args, **kwargs)
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/tests/chipyaml/chiptool.py", line 106, in chiptool_py
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :     success = send_yaml_command(chiptool, commands[1], server_path, server_arguments, show_adapter_logs, specifications_paths, pics,
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :     return f(get_current_context(), *args, **kwargs)
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.258 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/tests/chipyaml/tests_tool.py", line 45, in send_yaml_command
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return ctx.forward(test_tool)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 804, in forward
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return __self.invoke(__cmd, *args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return __callback(*args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/decorators.py", line 33, in new_func
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return f(get_current_context(), *args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/tests/chipyaml/runner.py", line 362, in chiptool
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return ctx.forward(websocket)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 804, in forward
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return __self.invoke(__cmd, *args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return __callback(*args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/decorators.py", line 92, in new_func
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return ctx.invoke(f, obj, *args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/.environment-vscode/pigweed-venv/lib/python3.12/site-packages/click/core.py", line 783, in invoke
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return __callback(*args, **kwargs)
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/tests/chipyaml/runner.py", line 334, in websocket
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :     return loop.run_until_complete(runner.run(parser_group.builder_config, runner_config))
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.074 - TEST ERR  :   File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     return future.result()
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :            ^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/py_matter_yamltests/matter/yamltests/runner.py", line 151, in run
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     raise (result)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/py_matter_yamltests/matter/yamltests/runner.py", line 218, in _run
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     responses, logs = await config.pseudo_clusters.execute(request, parser.definitions)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/py_matter_yamltests/matter/yamltests/pseudo_clusters/pseudo_clusters.py", line 55, in execute
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     status = await command(request)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :              ^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/py_matter_yamltests/matter/yamltests/pseudo_clusters/clusters/system_commands.py", line 90, in FactoryReset
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     AccessoryServerBridge.factoryReset(request)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/workspaces/connectedhomeip/scripts/py_matter_yamltests/matter/yamltests/pseudo_clusters/clusters/accessory_server_bridge.py", line 115, in factoryReset
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     proxy.factoryReset(register_key)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 1122, in __call__
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     return self.__send(self.__name, args)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 1461, in __request
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     response = self.__transport.request(
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :                ^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 1166, in request
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     return self.single_request(host, handler, request_body, verbose)
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 1182, in single_request
2025-10-29 16:35:26.259 ERROR   16:35:26.075 - TEST ERR  :     return self.parse_response(resp)
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :            ^^^^^^^^^^^^^^^^^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 1351, in parse_response
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :     return u.close()
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :            ^^^^^^^^^
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :   File "/usr/lib/python3.12/xmlrpc/client.py", line 668, in close
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  :     raise Fault(**self._stack[0])
2025-10-29 16:35:26.259 ERROR   16:35:26.076 - TEST ERR  : xmlrpc.client.Fault: <Fault 1: "<class 'Exception'>:Stopped subprocess terminated abnormally">
2025-10-29 16:35:26.259 ERROR   ================ CAPTURED LOG END ====================
2025-10-29 16:35:26.259 ERROR   Subprocess failed with exit code: -15
2025-10-29 16:35:26.291 ERROR   TestSystemCommands             - FAILED in 9.74 seconds
Traceback (most recent call last):
  File "/workspaces/connectedhomeip/./scripts/tests/run_test_suite.py", line 458, in cmd_run
    test.Run(
  File "/workspaces/connectedhomeip/scripts/tests/chiptest/test_definition.py", line 494, in Run
    runner.RunSubprocess(
  File "/workspaces/connectedhomeip/scripts/tests/chiptest/runner.py", line 167, in RunSubprocess
    raise Exception('Command %r failed: %d' % (cmd, s.returncode))
Exception: Command ['ip', 'netns', 'exec', 'tool', 'python3', '/workspaces/connectedhomeip/scripts/tests/chipyaml/chiptool.py', 'tests', 'TestSystemCommands', '--PICS', 'src/app/tests/suites/certification/ci-pics-values', '--server_path', './out/linux-x64-chip-tool-/chip-tool', '--server_arguments', 'interactive server --interface-id -1 --storage-directory /tmp/tmp9nzh21xr'] failed: 1
```

I propose to add a short delay between those operations in order to allow the app to setup signal handlers properly. I judge this situation to be an artifact of the testing framework together with my workstation setup and I do not think we can improve this in the SDK.

EDIT:
The final solution was different than what was initially propsed. Please scroll down for explanation.

#### Testing

Manual test runs on the TestSystemCommands. Before the change 1 in 3 failures. After the change 10 successful attempts.
#### Readability checklist

The checklist below will help the reviewer finish PR review in time and keep the
code readable:

-   [X] PR title is
        [descriptive](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html#title-formatting)
-   [X] Apply the
        [_“When in Rome…”_](https://project-chip.github.io/connectedhomeip-doc/style/CODING_STYLE_GUIDE.html)
        rule (coding style)
-   [X] PR size is short
-   [X] Try to avoid "squashing" and "force-update" in commit history
-   [ ] CI time didn't increase

I time will increase by 1 second per YAML tests run.

See: [Pull Request Guidelines](https://project-chip.github.io/connectedhomeip-doc/contributing/pull_request_guidelines.html)

enkiusz committed:
TestSystemCommands: Delay between factory resets to prevent sending a SIGTERM before signal handler is setup

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Add a new log message and watch for it instead of simply delaying

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

enkiusz commented:
A different approach has been implemented instead of a simple wait. Now a message is printed just before the event loop is started in Linux platform code. When this message is seen by the chipyaml runner it can be sure that the application has completely initialized.

enkiusz committed:
Test Darwin YAML breakage

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Increase timeout

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Fix timeout parameter name, more debug

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Commissioning timeout should not have been replaced

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
debug CapturedLogContains

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Waiting for 'Start event loop' won't work as App.start() already waits for 'mDNS service published:' which happens in the event loop

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Make application up detection more robust by checking both mDNS advertisement and event loop start

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Merge remote-tracking branch 'upstream/master' into delay-between-factoryresets

enkiusz committed:
Remove debug spam.
Mark symptom log messages as such in SDK source code.
Properly format exception strings.

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Merge branch 'master' into delay-between-factoryresets

enkiusz committed:
Restore workflow files deleted by mistake

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Explain improved application startup detection logic

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by clang-format

restyled-commits committed:
Restyled by autopep8

enkiusz committed:
Improve readability, fix exception string formatting

Signed-off-by: Maciej Grela <m.grela@samsung.com>

andy31415 commented on scripts/tests/chiptest/test_definition.py:
```diff
@@ -142,30 +145,42 @@ def __startServer(self):
+        """
+        server_process = kwargs.get('server_process', self.process)
+        outpipe = kwargs.get('outpipe', self.outpipe)
+        timeoutInSeconds = kwargs.get('timeoutInSeconds', 10)
```
Should we maybe kave kwargs be something like `server_process=None, outpipe=None` and such? 

If we use full **kwargs, this is really hard to self-document.
Also can we have symptoms be `list[str]` and pass it as such? That way looking at the signature is self-describing and static linters will check types.

andy31415 commented on scripts/tests/chiptest/test_definition.py:
```diff
@@ -142,30 +145,42 @@ def __startServer(self):
+        symptom_dict = {}
+
+        def searchSymptoms():
+            for s in symptoms:
```
We seem to iterate over everything anyway. Can we re-write this to be easier to follow (less dicts and list compression) by the logic of `if first not found, then we did not find`? Thinking something like (untested):

```python
def logHasAllExpectations():
   maxLogIndex = self.lastLogIndex
   for s in expectations:
        found, index = outpipe.CapturedLogContains(s, self.lastLogIndex)
        if not found:
            return False
        maxLogIndex = max(maxLogIndex, index)
  self.lastLogIndex = maxLogIndex + 1
  return True
```

enkiusz committed:
Merge branch 'master' of https://github.com/project-chip/connectedhomeip into delay-between-factoryresets

enkiusz committed:
Implement post-review changes

- add type hints, use explicit arguments instead of *args and **kwargs
- make search strategy more efficient, improve readability

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by isort

enkiusz committed:
Update test_definition.py

Force lists, fix off-by-one

arkq commented on scripts/tests/chiptest/test_definition.py:
```diff
@@ -142,30 +148,43 @@ def __startServer(self):
 
-    def __waitFor(self, waitForString, server_process, outpipe, timeoutInSeconds=10):
-        logging.debug('Waiting for %s' % waitForString)
+    def __waitFor(self, patterns: Iterable[str], process=None, outpipe: LogPipe | None = None, timeoutInSeconds: int = 10):
```
Since `process` and `outpipe` are used from the `self` in all use cases (if I'm not mistaken) these method's args are not required at all. After all the `lastLogIndex` is kept in the self, so providing external process instance seems not right.

arkq commented on examples/platform/linux/AppMain.cpp:
```diff
@@ -1017,6 +1017,9 @@ void ChipLinuxAppMainLoop(AppMainLoopImplementation * impl)
     sigaction(SIGTERM, &sa, nullptr);
 #endif
 
+    // This message is used as a symptom for when the application process has started.
```
I would change the "symptom" to something else because it sounds like "something is not right and we are detecting symptoms". Also indicate were to look at (some method like `App.waitForApplicationUp()` or something else).

enkiusz committed:
Post-review changes:

- use self.process and self.outpipe in chiptest.App.__WaitFor
- use 'marker' instead of 'symptom'

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by clang-format

restyled-commits committed:
Restyled by ruff

enkiusz committed:
minimalmdns: symptom -> marker

Signed-off-by: Maciej Grela <m.grela@samsung.com>

restyled-commits committed:
Restyled by isort

enkiusz committed:
Leftover symptom -> marker

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz committed:
Merge branch 'master' into delay-between-factoryresets

enkiusz committed:
Merge branch 'master' into delay-between-factoryresets

enkiusz committed:
Merge branch 'master' into delay-between-factoryresets

andy31415 commented on examples/platform/linux/AppMain.cpp:
```diff
@@ -1017,6 +1017,9 @@ void ChipLinuxAppMainLoop(AppMainLoopImplementation * impl)
 
+    // This message is used as a marker for when the application process has started.
+    // See: scripts/tests/chiptest/test_definition.py
+    ChipLogProgress(DeviceLayer, "Starting event loop");
```
nit: if we do this maybe we should make it a bit more magical, like `===== APP STATUS: Starting event loop =====` or similar, to avoid the chance of others writing similar messages.

Beyond that, should we add some TODO that for linux we should use some other messaging method? We are looking at doing this for OTA and stop log scraping, yet here we are again log scraping.

I am find as a followup, however in practice I believe we should have something more stable, since we are on linux for example we could use named pipes (and then tests can wait for named pipe appearance and then use that) or lock files that we give through command line or any other real IPC mechanism.

enkiusz committed:
Post-review fixes

Signed-off-by: Maciej Grela <m.grela@samsung.com>

enkiusz commented:
Sorry about that, I will improve communication.
